### ISSUE TITLE
List of emails on the results email modal isn't a real list

### ISSUE THREAD

**Issue by emmastephenson** [CDCgov/prime-simplereport#4088]
## Area or Component
Results page - email modal ("Email results")

## Quick Summary
The list of emails is not coded properly as a list of items.

## Accessibility Violation
Visual list is not marked up as list: https://cdc-devtools.dequecloud.com/rule-help/en/semantic-list

## Action Requested
Change the DOM to include a real list for emails in the modal.

## Screenshots (optional)
![Screen Shot 2022-07-26 at 4 37 17 PM](https://user-images.githubusercontent.com/80282552/181130134-ec43bcd5-e90b-4b89-a5a1-45b1fba254d3.png)

## Original Deque Issue
https://docs.google.com/spreadsheets/d/1b6xjubAN5Uw11DV4_B5zvSu7HeDI6Z-DpV4DMcToqAY/edit#gid=1978534823&range=33:33
